src/storage-types/src/sources/mysql.rs

```diff
@@ -159,3 +167,139 @@ impl RustType<ProtoMySqlSourceDetails> for MySqlSourceDetails {
+            TransactionId::Active(id) => write!(f, "{}", id),
+        }
+    }
+}
+
+impl From<TransactionId> for u64 {
+    fn from(id: TransactionId) -> Self {
+        match id {
+            // TransactionIds start at 1, so we can use 0 to represent the absent id
+            TransactionId::Absent => 0,
+            TransactionId::Active(id) => id,
+        }
+    }
+}
+
+impl TransactionId {
+    pub const MAX: TransactionId = TransactionId::Active(u64::MAX);
+}
+
+impl Add for TransactionId {
+    type Output = Self;
+
+    fn add(self, other: Self) -> Self {
+        match self {
+            TransactionId::Absent => other,
+            TransactionId::Active(id) => match other {
+                TransactionId::Absent => self,
+                TransactionId::Active(other_id) => TransactionId::Active(id + other_id),
+            },
+        }
+    }
+}
+
+impl Timestamp for TransactionId {
+    // No need to describe complex summaries
+    type Summary = ();
+
+    fn minimum() -> Self {
+        TransactionId::Absent
+    }
+}
+
+impl From<&TransactionId> for u64 {
+    fn from(id: &TransactionId) -> Self {
+        match id {
+            // TransactionIds start at 1, so we can use 0 to represent the absent id
+            TransactionId::Absent => 0,
+            TransactionId::Active(id) => *id,
+        }
+    }
+}
+
+impl TotalOrder for TransactionId {}
+
+impl PartialOrder for TransactionId {
+    fn less_equal(&self, other: &Self) -> bool {
+        Into::<u64>::into(self).less_equal(&Into::<u64>::into(other))
+    }
+}
+
+impl PathSummary<TransactionId> for () {
+    fn results_in(&self, src: &TransactionId) -> Option<TransactionId> {
+        Some(*src)
+    }
+
+    fn followed_by(&self, _other: &Self) -> Option<Self> {
+        Some(())
+    }
+}
+
+impl Refines<()> for TransactionId {
+    fn to_inner(_other: ()) -> Self {
+        Self::minimum()
+    }
+
+    fn to_outer(self) -> () {}
+
+    fn summarize(_path: Self::Summary) -> <() as Timestamp>::Summary {}
+}
+
+/// This type is used to track the progress of the ingestion dataflow, and represents
+/// a 'Gtid Set' of transactions that have been committed on a MySQL cluster.
+/// Each Gtid corresponds to a specific committed transaction, and contains the SourceId (a UUID)
+/// of the server that committed the transaction, and the TransactionId (a u64) representing
+/// the transaction's position.
+/// This type partitions by SourceId and contains the highest TransactionId committed for each
+/// SourceId 'range' in the Gtid Set. We use the highest TransactionId rather than a set of intervals
+/// because we validate that the MySQL server has been configured maintain ordering of transactions
+/// within a source, and that the TransactionIds are monotonically increasing.
+/// A None TransactionId represents that no transactions have been seen for the corresponding
+/// sources (which is useful to represent a frontier of all future possible sources)
```

Comment:
Feels like this doc comment needs some wordsmithing to improve clarity. The `None` variant is now named `Absent`. Also, this type doesn't represent a 'Gtid Set`, but a single element in such set. The full set is represented by an Antichain of this type.